kernel/src/utilities/static_ref.rs

```diff
@@ -25,6 +25,9 @@ impl<T> StaticRef<T> {
     pub const unsafe fn new(ptr: *const T) -> StaticRef<T> {
         StaticRef { ptr: ptr }
     }
+    pub const fn for_test(r: &T) -> StaticRef<T> {
+        StaticRef { ptr: r as *const T }
+    }
```

Comment:
I think this allows you to do something like:

```rust
let static_ref = {
    let refs = UartRegisters::expectable();
    StaticRef::for_test(&regs)
};
// static_refs is now a dangling pointer
let uart = Uart::new(static_ref, 10_000_000);
```

causing undefined behavior. Am I missing something?

Reply:
Yes, this is currently UB.

To fix this, we probably need to pass an `Rc<T>` to the `for_test` function and stash it in a global hashmap keyed by the pointer value.  Then, when the `StaticRef` is dropped, manage the state of the hashmap.